aioredis/cluster/cluster.py

```diff
@@ -0,0 +1,615 @@
+            node,
+            db=self._db,
+            password=self._password,
+            encoding=self._encoding,
+            commands_factory=self._factory,
+            loop=self._loop
+        )
+        try:
+            nodes_raw_resp = yield from conn.cluster_nodes()
+            return nodes_raw_resp
+        finally:
+            conn.close()
+            yield from conn.wait_closed()
+
+    @asyncio.coroutine
+    def fetch_cluster_info(self):
+        logger.info('Loading cluster info from {}...'.format(self._nodes))
+        tasks = [async_task(self._get_raw_cluster_info_from_node(node),
+                            loop=self._loop) for node in self._nodes]
+
+        try:
+            for task in asyncio.as_completed(tasks, loop=self._loop):
+                try:
+                    nodes_raw_response = yield from task
+                    self._cluster_manager = ClusterNodesManager.create(
+                        nodes_raw_response)
+                    logger.info('Cluster info loaded successfully: %s',
+                                nodes_raw_response)
+                    return
+                except (ReplyError, ProtocolError, ConnectionError) as exc:
+                    logger.warning(
+                        "Loading cluster info from a node failed with {}"
+                        .format(repr(exc))
+                    )
+        finally:
+            for task in tasks:
+                task.cancel()
+            # Wait until all tasks have closed their connection
+            yield from asyncio.gather(
+                *tasks, loop=self._loop, return_exceptions=True)
+
+        raise RedisClusterError(
+            "No cluster info could be loaded from any host")
+
+    @asyncio.coroutine
+    def initialize(self):
+        logger.info('Initializing cluster...')
+        self._moved_count = 0
+        yield from self.fetch_cluster_info()
+        logger.info('Initialized cluster.\n{}'.format(self._cluster_manager))
+
+    @asyncio.coroutine
+    def clear(self):
+        pass  # All connections are created on demand and destroyed afterwards.
+
+    @asyncio.coroutine
+    def create_connection(self, address):
+        conn = yield from create_redis(
+            address, db=self._db, encoding=self._encoding,
+            password=self._password, loop=self._loop, )
+        return conn
+
+    @asyncio.coroutine
+    def _execute_node(self, address, command, *args, **kwargs):
+        """Execute redis command and returns Future waiting for the answer.
+
+        :param command str
+        :param pool obj
+        Raises:
+        * TypeError if any of args can not be encoded as bytes.
+        * ReplyError on redis '-ERR' responses.
+        * ProtocolError when response can not be decoded meaning connection
+          is broken.
+        """
+        cmd = decode(command).lower()
+        to_close = []
+        try:
+            conn = yield from self.create_connection(address)
```

Comment:
The main reason is that I do not want to use a pool here. Typically, users will want to use RedisPoolCluster directly, where connections are stored in a pool.

I will wait for your pool refactoring and then merge RedisCluster and RedisPoolCluster into a single class if it is still relevant.